packages/pigeon/lib/objc_generator.dart

```diff
@@ -92,8 +92,16 @@ const Map<String, String> _propertyTypeForDartTypeMap = <String, String>{
   'Map': 'strong',
 };
 
-String? _objcTypeForDartType(String type) {
-  return _objcTypeForDartTypeMap[type];
+String? _objcTypeForPrimitiveDartType(String type) {
+  return _objcTypeForDartTypeMap.containsKey(type)
+      ? '${_objcTypeForDartTypeMap[type]} *'
+      : null;
+}
+
+String _objcTypeForDartType(String? classPrefix, String type) {
+  return _objcTypeForDartTypeMap.containsKey(type)
+      ? _objcTypeForDartTypeMap[type]!
+      : _className(classPrefix, type);
```

Comment:
You can write this without `!` (making it safer against future mistakes, such as if `type` were changed in one line but not the other) as:
```
final String? objcType = _objcTypeForDartTypeMap[type];
return objcType ?? _className(classPrefix, type);
```

Reply:
I don't actually prefer this syntax since I think it hides the branch.  I didn't realize that until I already did it.  It's not a huge deal so I'll keep it changed like you mentioned.

Reply:
You could write it with an explicit branch if you prefer:
```
final String? objcType = _objcTypeForDartTypeMap[type];
if (objcType != null) {
   return objcType;
}
return _className(classPrefix, type);
```

But it is objectively better to write code that has enforced type safety, rather than code that bypasses type safety and relies on people to not make mistakes.